contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -519,202 +554,430 @@ contract VaultHub is PausableUntilWithRoles {
-        uint256 amountToRebalance = Math256.min(maxAmountToRebalance, _vault.balance);
+    /// @dev msg.sender should be vault's owner
+    function pauseBeaconChainDeposits(address _vault) external {
+        _checkConnectionAndOwner(_vault);
 
-        // TODO: add some gas compensation here
-        IStakingVault(_vault).rebalance(amountToRebalance);
+        IStakingVault(_vault).pauseBeaconChainDeposits();
     }
 
-    /// @notice rebalances the vault by writing off the amount of ether equal
-    ///     to `msg.value` from the vault's liability stETH
-    /// @dev msg.sender should be vault's contract
-    function rebalance() external payable whenResumed {
-        if (msg.value == 0) revert ZeroArgument("msg.value");
+    /// @notice resumes beacon chain deposits for the vault
+    /// @param _vault vault address
+    /// @dev msg.sender should be vault's owner
+    function resumeBeaconChainDeposits(address _vault) external {
+        VaultConnection storage connection = _checkConnectionAndOwner(_vault);
+        if (!_isVaultHealthy(connection, _storage().records[_vault])) revert UnhealthyVaultCannotDeposit(_vault);
 
-        VaultSocket storage socket = _connectedSocket(msg.sender);
+        IStakingVault(_vault).resumeBeaconChainDeposits();
+    }
 
-        uint256 sharesToBurn = LIDO.getSharesByPooledEth(msg.value);
-        uint256 liabilityShares = socket.liabilityShares;
-        if (liabilityShares < sharesToBurn) revert InsufficientSharesToBurn(msg.sender, liabilityShares);
+    /// @notice Emits a request event for the node operator to perform validator exit
+    /// @param _vault vault address
+    /// @param _pubkeys array of public keys of the validators to exit
+    /// @dev msg.sender should be vault's owner
+    function requestValidatorExit(address _vault, bytes calldata _pubkeys) external {
+        _checkConnectionAndOwner(_vault);
 
-        socket.liabilityShares = uint96(liabilityShares - sharesToBurn);
+        IStakingVault(_vault).requestValidatorExit(_pubkeys);
+    }
 
-        LIDO.rebalanceExternalEtherToInternal{value: msg.value}();
+    /// @notice Triggers validator exit for the vault using EIP-7002
+    /// @param _vault vault address
+    /// @param _pubkeys array of public keys of the validators to exit
+    /// @dev msg.sender should be vault's owner
+    function triggerValidatorWithdrawals(
+        address _vault,
+        bytes calldata _pubkeys,
+        uint64[] calldata _amounts,
+        address _refundRecipient
+    ) external payable {
+        VaultConnection storage connection = _checkConnectionAndOwner(_vault);
+        VaultRecord storage record = _storage().records[_vault];
+
+        // disallow partial validator withdrawals when the vault is unhealthy,
+        // in order to prevent the vault owner from jamming the consensus layer withdrawal queue
+        // delaying the forceful validator exits required for rebalancing the vault
```

Comment:
```suggestion
        // disallow partial validator withdrawals when the vault is unhealthy,
        // in order to prevent the vault owner from clogging the consensus layer withdrawal queue
        // front-running and delaying the forceful validator exits required for rebalancing the vault
```